test/parallel_api/algorithm/alg.nonmodifying/gather.pass.cpp

```diff
@@ -0,0 +1,80 @@
+// -*- C++ -*-
+//===-- gather.pass.cpp ----------------------------------------------------===//
```

Comment:
1) The tests should be compiled not-SYCL compilator as well.
2) Also the test should be designed in the same manner as the other PSTL tests.